WcaOnRails/app/models/registration.rb

```diff
@@ -226,6 +226,11 @@ def self.wcif_json_schema
   private def user_can_register_for_competition
     if user&.cannot_register_for_competition_reasons.present?
       errors.add(:user_id, user.cannot_register_for_competition_reasons.to_sentence)
+    elsif user.banned?
+      ban_end = user.current_team_members.select(:name == "Banned Competitors").first.end_date
```

Comment:
I think using the team name here is quite risky, I'd rather use a more reliable way like:
```suggestion
      ban_end = user.current_team_members.select { |m| m.team == Team.banned }.first.end_date
```
I didn't actually run the code so I may have messed up something, but hopefully you get my point :)

Reply:
thanks, implemented